promql/functions.go

```diff
@@ -765,6 +807,13 @@ var functions = map[string]*Function{
 		ReturnType: model.ValVector,
 		Call:       funcLabelReplace,
 	},
+	"label_join": {
+		Name:       "label_join",
+		ArgTypes:   []model.ValueType{model.ValVector, model.ValString, model.ValString, model.ValString, model.ValString, model.ValString},
```

Comment:
So far you added the new `Variadic` field to the struct but it is not being used anywhere.
What we need is to consider it in the type-validation stage of the parser: https://github.com/prometheus/prometheus/blob/master/promql/parse.go#L1075-L1085

The "unknown error" comes probably from an unbounded slice access happening because of the missing check. You should actually see more details in the logs, as the causing stack trace is printed: https://github.com/prometheus/prometheus/blob/master/promql/parse.go#L326-L341